code/game/objects/items/robot/robot_parts.dm

```diff
@@ -139,6 +139,11 @@
 
 /obj/item/robot_suit/attackby(obj/item/W, mob/user, params)
 
+	if(istype(W, /obj/item/mmi/posibrain/soul_vessel/agent))
+		var/obj/item/mmi/posibrain/soul_vessel/agent/buscheck = W
+		if(buscheck.agent) //if for some stupid reason team badmin wants to let people add one of these to a borg shell
+			to_chat(user, "<span class='warning'>[W] hums fitfully, but doesn't mesh with the frame.</span>")
+			return
```

Comment:
Theres no real reason for check for the specific path instead of the variable

```suggestion
	if(istype(W, /obj/item/mmi/posibrain/soul_vessel))
		var/obj/item/mmi/posibrain/soul_vessel/buscheck = W
		if(buscheck.agent) 
			to_chat(user, "<span class='warning'>[W] hums fitfully, but doesn't mesh with the frame.</span>")
			return
```